### PR TITLE
Add chaos mesh (chaos-mesh.org) client extension v1.0.1 (api v1alpha1)

### PR DESCRIPTION
**Pull request by hypnoce**
## Description

[Chaos Mesh](https://chaos-mesh.org) has recently moved to [GA](https://chaos-mesh.org/blog/chaos-mesh-1.0-chaos-engineering-on-kubernetes-made-easier).
The client extension exposes the :

- [IoChaos](https://chaos-mesh.org/docs/chaos_experiments/iochaos_experiment/)
- [NetworkChaos](https://chaos-mesh.org/docs/chaos_experiments/networkchaos_experiment/)
- [PodChaos](https://chaos-mesh.org/docs/chaos_experiments/podchaos_experiment/)
- PodNetworkChaos
- [StressChaos](https://chaos-mesh.org/docs/chaos_experiments/stresschaos_experiment/)
- [TimeChaos](https://chaos-mesh.org/docs/chaos_experiments/timechaos_experiment/)
- [KernelChaos](https://chaos-mesh.org/docs/chaos_experiments/kernelchaos_experiment/)

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by rohanKanojia** [comment]
This projects uses Apache 2.0 License. You would need to add license headers using `mvn -N license:format`

File: extensions/chaosmesh/client/pom.xml
Comment (oscerd) [inline extensions/chaosmesh/client/pom.xml:125]: I guess it should be Chaos Mesh :: Bom

File: extensions/chaosmesh/client/pom.xml
Comment (oscerd) [inline extensions/chaosmesh/client/pom.xml:132]: It should be Chaos Mesh :: Bom with Dependencies

**Review by oscerd — CHANGES_REQUESTED** [review]
Just a couple of typo. It seems to be ok. 

At some point I guess we may need to host extensions in a separate repository, because they could explode in terms of number.

**Comment by hypnoce** [comment]
@oscerd I think the sonar analysis is a bit wrong. Running locally shows coverage data.

**Comment by hypnoce** [comment]
Let me know if you need more work from my side. Thanks !

**Comment by manusa** [comment]
> @oscerd I think the sonar analysis is a bit wrong. Running locally shows coverage data (not 100% coverage though).

Coverage is calculated only for **unit** tests within the same module.

**Comment by zhouqiang-cl** [comment]
@hypnoce @manusa Chaos Mesh maintainer here, thank you very much for this PR. If need any assistance from Chaos Mesh team, please let me know

Thank you again

**Review by oscerd — APPROVED** [review]

File: extensions/chaosmesh/examples/src/main/java/io/fabric8/chaosmesh/examples/CreateIoChaos.java
Comment (iocanel) [inline extensions/chaosmesh/examples/src/main/java/io/fabric8/chaosmesh/examples/CreateIoChaos.java:40]: +1

**Review by iocanel — APPROVED** [review]

**Comment by rohanKanojia** [comment]
@hypnoce: Thanks a lot for your PR. It looks really good to be merged. Are you(or someone from your team) planning to write a blogpost about this extension whenever it gets merged?

**Comment by hypnoce** [comment]
> @hypnoce: Thanks a lot for your PR. It looks really good to be merged. Are you(or someone from your team) planning to write a blogpost about this extension whenever it gets merged?

@rohanKanojia do you have any blog in mind ?
If there is there anything for this PR to be merged, let me know. Thanks

**Comment by rohanKanojia** [comment]
@hypnoce: Sorry for not being clear. I was asking if you plan to write a blogpost about this Chaos Mesh extension. I wrote two blog posts about Knative[0] and Tekton[1] extensions few months ago. Maybe you can write a similar one for this one?

Regarding getting this PR merged, it's pending on review from  @manusa . Whenever he approves it, it will be merged.

[0] https://rohaan.medium.com/accessing-knative-rest-api-using-fabric8-knative-client-443a16ac43f7
[1] https://itnext.io/access-tekton-pipelines-in-java-using-fabric8-tekton-client-bd727bd5806a

**Review by manusa — APPROVED** [review]

**Review by rohanKanojia — APPROVED** [review]

**Comment by rohanKanojia** [comment]
@hypnoce : Hi, Could you please rebase this branch with master? We have made some breaking changes in kubernetes Client DSL #2566 .

**Comment by hypnoce** [comment]
> @hypnoce : Hi, Could you please rebase this branch with master? We have made some breaking changes in kubernetes Client DSL #2566 .

done

**Comment by hypnoce** [comment]
@rohanKanojia anything else I can do ?

**Comment by rohanKanojia** [comment]
@hypnoce : Hi, Sorry for the delay. Could you please modify the entry in CHANGELOG and move it to `5.0-SNAPSHOT` section? I think we can proceed with merge.

**Comment by rohanKanojia** [comment]
[merge]

**Comment by rohanKanojia** [comment]
[merge]

**Pull request merged**
